src/Symfony/Component/Console/Application.php

```diff
@@ -745,10 +726,14 @@ public function getTerminalDimensions()
      * @param int $height The height
      *
      * @return Application The current application
+     *
+     * @deprecated since version 3.2, to be removed in 4.0. Set the ANSICON env var instead.
      */
     public function setTerminalDimensions($width, $height)
     {
-        $this->terminalDimensions = array($width, $height);
+        @trigger_error(sprintf('%s is deprecated as of 3.2 and will be removed in 4.0. Set the ANSICON env var instead.', __METHOD__, ArgumentResolverInterface::class), E_USER_DEPRECATED);
+
+        putenv('ANSICON='.$width.'x'.$height);
```

Comment:
We should support a different env var following the same format if we want to go this way (but keeping support for the ANSICON one for auto-detection)